core/src/main/java/io/questdb/griffin/SqlOptimiser.java

```diff
@@ -3801,6 +3801,104 @@ private ExpressionNode rewriteAggregate(ExpressionNode agg, QueryModel model) {
         return agg;
     }
 
+    /**
+     * Rewrites
+     * SELECT LAST(timestamp) FROM table_name;
+     * AND
+     * SELECT max(timestamp) FROM table_name;
+     * into query which can is optimised search
+     * SELECT timestamp from (SELECT timestamp from table_name order by timestamp) limit 1
+     * This enables FrameBackwardScan for aggregation queries containing LAST keyword
+     * AND
+     * Rewrites
+     * SELECT FIRST(timestamp) FROM table_name;
+     * AND
+     * SELECT min(timestamp) FROM table_name;
+     * into query which can is optimised search
+     * SELECT timestamp from (SELECT timestamp from table_name) limit 1
+     * This disables invoking group by workers
+     */
+    private void rewriteGroupByForFirstLastMaxMinAggregateFunctions(QueryModel parent) {
+
+        //base condition to stop recursion
+        if (parent == null || parent.getNestedModel() == null)
+            return;
+
+        ObjList<QueryColumn> queryColumns = parent.getBottomUpColumns();
+        QueryModel nestedModel = parent.getNestedModel();
```

Comment:
Apart from the nested model there may be union and join models, so the rewrite should check them too. See `rewriteCountDistinct`. Also, it'll be great to have a test(s) around rewrite being applied to union / join subqueries.